lib/veritaserum.ex

```diff
@@ -27,44 +27,69 @@ defmodule Veritaserum do
       3
   """
   @spec analyze(String.t) :: Integer.t
-  def analyze(input) do
-    input
-    |> clean
-    |> String.split
-    |> split_on_emoticons
-    |> Enum.map(&String.trim/1)
-    |> analyze_list()
-    |> Enum.reduce(0, &(&1 + &2))
-  end
 
-  defp analyze_list([head | tail]) do
-    analyze_list(tail, head, [analyze_word(head)])
+  def analyze(input, options \\ [])
+  def analyze(input, options) do
+    return = Keyword.get(options, :return, :score)
+
+    marked_list = input
+                  |> clean
+                  |> String.split
+                  |> split_on_emoticons
+                  |> mark_list
+                  |> Enum.reverse
+
+
+    score = marked_list
+            |> analyze_list
+            |> Enum.reduce(0, &(&1 + &2))
+
+    case return do
+      :score -> score
+      :score_and_marks -> {score, marked_list}
+    end
   end
-  defp analyze_list([head | tail], previous, result) do
-    analyze_list(tail, head, [analyze_word(head, previous) | result])
+
+  defp mark_word(word) do
+    with {_, nil, _} <- {:negator,  Evaluator.evaluate_negator(word),  word},
+         {_, nil, _} <- {:booster,  Evaluator.evaluate_booster(word),  word},
+         {_, nil, _} <- {:emoticon, Evaluator.evaluate_emoticon(word), word},
+         {_, nil, _} <- {:word,     Evaluator.evaluate_word(word),     word},
```

Comment:
IMO this would look better without the weird indentations within the triples, just regular spaces like `{:emoticon, Evaluator.evaluate_emoticon(word), word}`